nixos/modules/services/networking/godns.nix

```diff
@@ -0,0 +1,87 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    types
+    ;
+
+  cfg = config.services.godns;
+
+  configFormat = pkgs.formats.yaml { };
+  configFile =
+    if cfg.configFile == null then configFormat.generate "config.yaml" cfg.config else cfg.configFile;
+in
+{
+
+  options.services.godns = {
+    enable = mkEnableOption "GoDNS service";
+
+    package = mkPackageOption pkgs "godns" { };
+
+    config = mkOption {
+      type = lib.types.submodule {
+        freeformType = configFormat.type;
+      };
+
+      default = { };
+
+      description = ''
+        Configuration for GoDNS. Refer to the [configuration section](1) in the
+        GoDNS GitHub repository for details.
+
+        [1]: https://github.com/TimothyYe/godns?tab=readme-ov-file#configuration
+      '';
+    };
+
+    configFile = lib.mkOption {
+      default = null;
+      description = ''
+        Path to a custom GoDNS configuration file.
+
+        If set, this option overrides the configuration provided by the `config`
+        option. This is particularly useful for specifying sensitive values like
+        the `login_token` for API providers. For generating a configuration file
+        with secrets, consider using [sops-nix templates][1].
+
+        [1]: https://github.com/Mic92/sops-nix?tab=readme-ov-file#templates
+      '';
+      type = types.nullOr types.path;
+    };
+
+    additionalRestartTriggers = mkOption {
```

Comment:
The systemd service will restart when the value of `cfg.configFile` changes. Is this really needed?

Reply:
When using something like `nix-sops` templates, the answer is sadly yes. Because the path of the generated template does not change even if the secret changes.

Reply:
> Because the path of the generated template does not change even if the secret changes.

How does this option help then?

Reply:
So sorry for the delay, as you can see here:
https://github.com/michaelvanstraten/systems/blob/8c016034f3028573c04340ea599cad378595fffe/modules/nixos/personal-cloud.nix#L8-L10
with `sops.nix` the content of the template changes when the template changes.